Buddies/Discover/DiscoverTableVC.swift

```diff
@@ -50,7 +59,14 @@ class DiscoverTableVC : ActivityTableVC {
     override func fetchAndLoadActivities() {
         self.startRefreshIndicator()
         let searchParams = searchBar.getSearchParams()
-        let topics = (searchParams.filterText?.isEmpty ?? true) ? getTopics() : []
+        let hasTextParam: Bool = !(searchParams.filterText?.isEmpty ?? true)
+        //If no topics and no query, hide results to show onboarding
+        guard hasTextParam || getTopics().count > 0 else {
+            self.updateWantedActivities(with: [[]])
```

Comment:
why are we clearing the data here if there is a search or there is no topics?

Reply:
If we aren't searching something, and we have no topics, this is when we want the onboarding message.

Reply:
Will add a comment ;)